tests/auxiliary/user-account/test_contract.py

```diff
@@ -0,0 +1,372 @@
+        (11, 9),
+        (UINT_256_MAX, 0),
+    ]
+    for min_block, max_block in invalid_block_ranges:
+        transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+            "nonce": get_nonce(vm),
+            "min_block": min_block,
+            "max_block": max_block,
+        }}).sign(PRIVATE_KEY)
+        computation = vm.apply_transaction(transaction)
+        assert computation.is_error
+
+    valid_block_ranges = [
+        (0, UINT_256_MAX),
+        (0, 10),
+        (10, 10),
+        (10, UINT_256_MAX)
+    ]
+    for min_block, max_block in valid_block_ranges:
+        transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+            "nonce": get_nonce(vm),
+            "min_block": min_block,
+            "max_block": max_block,
+        }}).sign(PRIVATE_KEY)
+        computation = vm.apply_transaction(transaction)
+        assert computation.is_success
+
+
+def test_call_transfers_value(vm):
+    with vm.state.state_db() as state_db:
+        balance_sender_before = state_db.get_balance(ACCOUNT_ADDRESS)
+        balance_destination_before = state_db.get_balance(DESTINATION_ADDRESS)
+
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": get_nonce(vm),
+        "value": 10
+    }}).sign(PRIVATE_KEY)
+    computation = vm.apply_transaction(transaction)
+    assert computation.is_success
+
+    with vm.state.state_db() as state_db:
+        balance_sender_after = state_db.get_balance(ACCOUNT_ADDRESS)
+        balance_destination_after = state_db.get_balance(DESTINATION_ADDRESS)
+
+    assert balance_sender_after == balance_sender_before - 10
+    assert balance_destination_after == balance_destination_before + 10
+
+
+def test_call_checks_signature(vm):
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": get_nonce(vm),
+    }}).sign(PRIVATE_KEY)
+
+    v, r, s = transaction.vrs
+    invalid_vrs_triples = [
+        (v + 1, r, s),
+        (v - 1, r, s),
+        (0, r, s),
+        (1, r, s),
+
+        (v, r + 1, s),
+        (v, r - 1, s),
+        (v, 0, s),
+
+        (v, r, s + 1),
+        (v, r, s - 1),
+        (v, r, 0),
+
+        (27 if v == 28 else 28, r, SECPK1_N - s)
+    ]
+
+    for invalid_vrs in invalid_vrs_triples:
+        transaction.vrs = invalid_vrs
+        assert transaction.is_signed
+
+        computation = vm.apply_transaction(transaction)
+        assert computation.is_error
+
+
+def test_call_uses_remaining_gas(vm):
+    transaction = ForwardingTransaction(**{**DEFAULT_TX_PARAMS, **{
+        "nonce": get_nonce(vm),
+        "destination": GAS_LOGGING_CONTRACT_ADDRESS,
+        "gas": 1 * 1000 * 1000,
+    }}).sign(PRIVATE_KEY)
+    computation = vm.apply_transaction(transaction)
+    assert computation.is_success
+
+    logs = computation.get_log_entries()
+    assert len(logs) == 1
+    assert big_endian_to_int(logs[0][-1]) > 900 * 1000  # some gas will have been consumed earlier
```

Comment:
A slightly easier to understand version of this assertion that I've used elsewhere is to assert that the delta is smaller than a certain value.

```
gas_used = big_endian_to_int(logs[0][-1])
assert transaction.gas - gas_used < 1000
```